components/contact/ContactField.tsx

```diff
@@ -0,0 +1,42 @@
+import { ContactFieldType } from '@/types';
+
+const ContactField = ({
+  label,
+  id,
+  type = 'text',
+  component = 'input',
+  getInputStyle,
+  register,
+  errors,
+}: ContactFieldType) => {
+  const inputProps = {
+    ...register(id),
+    type,
+    className: getInputStyle(errors[id]),
+  };
+
+  return (
+    <div className="max-w-[21.6rem] s:max-w-[35rem] md:max-w-[40rem] lg:max-w-[55rem]">
+      <label
+        htmlFor={id}
+        className="mb-4 block text-lg font-normal leading-[160%] text-black-300 dark:text-white-900 md:text-xl"
+      >
+        {label}
+      </label>
+
+      {component === 'input' ? (
```

Comment:
Another approach to consider that will improve readability is mapping the object or using a switch case. In this example, I am creating a componentMap that will render the input or the textarea... To implement this, you just need to render {Component} in your return statement.

```
const componentMap = {
  input: <input {...inputProps} />,
  textarea: <textarea style={{ resize: 'none' }} {...inputProps}></textarea>,
};

const Component = componentMap[component] || null;
```

What you have works fine, but the approach I provided seems a little clearer and cleaner... You can use either option here.